### ISSUE TITLE
Battery percentage not showing on traccar web

### ISSUE THREAD

**Issue by dmetrosoft** [vitalidze/traccar-web#788]
Please i am using traccar with TK05, every thing seems to work fine, but am having problem with the battery percentage, i checked on MySQL table and discovered that the IP and battery percentage is in the other column in position table, i don't know how to make the battery percentage show, please help me out. thanks

**Comment by vitalidze** [comment]
You can use [Sensors](http://traccar.litvak.su/features/sensors.html) functionality to display value of your battery. It will be shown in popup when you hover a row in table with devices at the left hand side and when you hover a marker on the map (both in archive and live data).

**Comment by dmetrosoft** [comment]
thanks, please how can i stop 0 log and lat position from inserting into MySQL database

**Comment by vitalidze** [comment]
In `traccar.xml` configuration there are options to configure positions filtering. See official documentation here: https://www.traccar.org/configuration-file/

These two should work for you:
- `filter.enable`   Boolean flag to enable or disable position filtering.
- `filter.zero` Filter zero coordinates. Zero latitude and longitude are theoretically valid values, but it practice it usually indicates invalid GPS data.

**Comment by dmetrosoft** [comment]
thanks for the reply, now the 0 has stopped but i am trying to get the battery percentage, i look at the position table in mysql database and discovered that the power column is null each time a new position is uploaded. Below is the details from log
HEX
283836343736383031303831353533352c5a4330372c3239303731362c412c303435302e35343437344e2c30303730322e3137353338452c302e3532382c3132343831302c3030302e3029283836343736383031303831353533352c5a4332302c3239303731362c3132343831302c342c3337312c36353533352c32353529
DECODED
(864768010815535,ZC07,290716,A,0450.54474N,00702.17538E,0.528,124810,000.0)(864768010815535,ZC20,290716,124810,4,371,65535,255)
the device is TK05 which uses TK103 protocol, thanks

**Comment by dmetrosoft** [comment]
[Revised Protocol 150306.pdf](https://github.com/vitalidze/traccar-web/files/391680/Revised.Protocol.150306.pdf)
attached here is the protocol documentation, thanks

**Comment by vitalidze** [comment]
The `power` column is not used by the traccar application at all. It is a subject to be removed in newer versions. You should look for the battery values in `other` column. It contains all additional data.

**Comment by MarceloMJM** [comment]
Taking advantage of the issue, the other column besides the text values in the case of ignition.
It presents true and false, it is possible the sensor configuration with this text (true - false)?

**Comment by vitalidze** [comment]
You cannot set up intervals for the text values, only for numbers. Other configuration is possible.

**Comment by Fleetracking** [comment]
Hi 
@vitalidze i use a tracker communicating with H02 protocol.
How can i extact different informations (ignation, battery,...) from "other" column (one String in DB).

Another question please, i tried to add "battery consumption" as a sensor but i don't know which "pamameter" to configure for thi sensor (other.X) ? 
I tried "power" as a parameter to check if this "battery" sensor  will be displayed in GUI but its'nt the case (i followed step by step "Sensors" page instructions). 

BR

**Comment by vitalidze** [comment]
@Fleetracking you need to check this out using your device documentation. The ignition, battery and so on parameters may have different codes (i.e. parameter names) depending on your device type. What does it show in device popup?

**Comment by Fleetracking** [comment]
Hi
@vitalidze : according to H02 protocol documentation i'm able to decode different codes, but the problem is that all these parameters are saved in one field in DB ("other" column).
So how can i extract one of these parameters from "other" field, and use it for example to configure a sensor?
Regarding your question, in device popup you can find (odometer, protocol, ignition, status, ip).
BR

**Comment by vitalidze** [comment]
> @vitalidze : according to H02 protocol documentation i'm able to decode different codes, but the problem is that all these parameters are saved in one field in DB ("other" column).

Yes, all of them are stored in one column. In web UI they are displayed in popup one-by-one.

> So how can i extract one of these parameters from "other" field, and use it for example to configure a sensor?

You should use values from popup (odometer, protocol, ignition, status, ...) as the value in "Parameter" column (see http://traccar.litvak.su/features/sensors.html), then you can configure visibility, intervals, colors and icons as said in official documentation: http://traccar.litvak.su/features/sensors.html
